tests/unittests/media/server/main/mediaPipeline/MiscellaneousFunctionsTest.cpp

```diff
@@ -243,6 +244,89 @@ TEST_F(RialtoServerMediaPipelineMiscellaneousFunctionsTest, GetPositionSuccess)
     EXPECT_EQ(targetPosition, m_kPosition);
 }
 
+/**
+ * Test that SetImmediateOutput returns failure if the gstreamer player is not initialized
+ */
+TEST_F(RialtoServerMediaPipelineMiscellaneousFunctionsTest, SetImmediateOutputFailureDueToUninitializedPlayer)
+{
+    mainThreadWillEnqueueTaskAndWait();
+    const int kSourceId{1};
+    EXPECT_FALSE(m_mediaPipeline->setImmediateOutput(kSourceId, true));
+}
+
+/**
+ * Test that SetImmediateOutput returns failure if the gstreamer API fails
+ */
+TEST_F(RialtoServerMediaPipelineMiscellaneousFunctionsTest, SetImmediateOutputFailure)
+{
+    loadGstPlayer();
+    int videoSourceId = attachSource(firebolt::rialto::MediaSourceType::VIDEO, "video/h264");
+    mainThreadWillEnqueueTaskAndWait();
+    EXPECT_CALL(*m_gstPlayerMock, setImmediateOutput(_, _)).WillOnce(Return(false));
+    EXPECT_FALSE(m_mediaPipeline->setImmediateOutput(videoSourceId, true));
+}
+
+/**
+ * Test that SetImmediateOutput returns success if the gstreamer API succeeds
+ */
+TEST_F(RialtoServerMediaPipelineMiscellaneousFunctionsTest, SetImmediateOutputSuccess)
+{
+    loadGstPlayer();
+    int videoSourceId = attachSource(firebolt::rialto::MediaSourceType::VIDEO, "video/h264");
+
+    mainThreadWillEnqueueTaskAndWait();
+    EXPECT_CALL(*m_gstPlayerMock, setImmediateOutput(_, true)).WillOnce(Return(true));
+    EXPECT_TRUE(m_mediaPipeline->setImmediateOutput(videoSourceId, true));
+
+    mainThreadWillEnqueueTaskAndWait();
+    EXPECT_CALL(*m_gstPlayerMock, setImmediateOutput(_, false)).WillOnce(Return(true));
+    EXPECT_TRUE(m_mediaPipeline->setImmediateOutput(videoSourceId, false));
+}
+
+/**
```

Comment:
Add tests for when source not found, can follow FlushNoSourcePresent. This is for both setImmediateOutput and getImmediateOutput, and if you could also add one for getStats as that's missing the same.